ckanext/stadtzhtheme/controller.py

```diff
@@ -0,0 +1,180 @@
+from urllib import urlencode
+import logging
+
+from ckan.plugins import toolkit as tk
+import ckan.model as model
+import ckan.logic as logic
+import ckan.lib.plugins
+from ckan.common import c, config, _, request, OrderedDict
+import ckan.lib.helpers as h
+import ckan.lib.search as search
+import ckan.lib.base as base
+from six import string_types
+
+
+import ckan.controllers.group as group
+
+lookup_group_controller = ckan.lib.plugins.lookup_group_controller
+log = logging.getLogger(__name__)
+get_action = logic.get_action
+NotFound = logic.NotFound
+NotAuthorized = logic.NotAuthorized
+ValidationError = logic.ValidationError
+abort = tk.abort
+render = base.render
+
+
+class OgdzhGroupSearchController(group.GroupController):
+    """
+    backporting [https://github.com/ckan/ckan/pull/4032]
+    form CKAN 2.8: Improved search in group read and bulk page
+    TODO: remove this controller overwrite after upgrade to CKAN 2.8
+
+    This controller is copied over from the groups.py controller
+    of CKAN version 2.8. As soon as the update to 2.8 has been
+    done this patch is not necessary any more.
+
+    This overwrite forces CKAN to use the 'fq' parameter
+    instead of the 'q' parameter for filtering by groups
+    in the solr search query.
+    """
+
+    def _read(self, id, limit, group_type): # noqa
+        ''' This is common code used by both read and bulk_process'''
+        context = {'model': model, 'session': model.Session,
+                   'user': c.user,
+                   'schema': self._db_to_form_schema(group_type=group_type),
+                   'for_view': True, 'extras_as_string': True}
+
+        q = c.q = request.params.get('q', '')
+
+        # backporting [#https://github.com/ckan/ckan/pull/4032]
+        # for group filterin the solr filter 'fq' should be used
+        # rather then th query parameter 'q', that has been used
+        # up to CKAN 2.6
```

Comment:
And here:
```
This overwrite forces CKAN to use the 'fq' parameter instead of the 'q' parameter for filtering by groups in the solr search query.
```